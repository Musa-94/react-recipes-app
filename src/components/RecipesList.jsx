import React, { PureComponent } from 'react';

class RecipesList extends PureComponent {
    render() {
        return (
            <div>
                Hello from Recipes List
            </div>
        );
    }
}

export default RecipesList;
