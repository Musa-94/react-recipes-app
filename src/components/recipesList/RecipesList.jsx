import React from 'react';
import SingleRecipe from '../../pages/SingleRecipe.jsx';

const RecipesList = props => {
    const { recipes } = props;

    return (
        <div className={'container py-5'}>
            {/* title */}
            <div>
                <div className='col-10 mx-auto col-md-6 text-center text-uppercase mb-3'>
                    <h1 className={'text-slanted'}> recipe list </h1>
                </div>
            </div>
            {/* end of title */}
            <div className={'row'}>
                {recipes.map((recipe, index) =>
                    <SingleRecipe
                        key={index}
                        title={recipe.title}
                        recipeId={recipe.recipe_id}
                        imageUrl={recipe.image_url}
                        sourceUrl={recipe.source_url}
                        publisher={recipe.publisher}
                    />
                )}
            </div>
        </div>
    );
}

export default React.memo(RecipesList);
