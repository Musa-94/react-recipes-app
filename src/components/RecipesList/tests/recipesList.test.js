import React from 'react';
import RecipesList from '../RecipesList.jsx';

const props = {
    recipes: [
        {
            image_url: "http://forkify-api.herokuapp.com/images/best_pizza_dough_recipe1b20.jpg",
            publisher: "101 Cookbooks",
            publisher_url: "http://www.101cookbooks.com",
            recipe_id: "47746",
            social_rank: 100,
            source_url: "http://www.101cookbooks.com/archives/001199.html",
            title: "Best Pizza Dough Ever",
        }
    ],
}

describe('RecipesList snapshot', () => {
    it('should correct render', () => {
        const component = renderer(<RecipesList {...props}/>);

        expect(component).toMatchSnapshot()
    })
})