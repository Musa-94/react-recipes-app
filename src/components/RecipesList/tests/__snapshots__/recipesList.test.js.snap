// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesList snapshot should correct render 1`] = `
<div
  className="container py-5"
>
  <div>
    <div
      className="col-10 mx-auto col-md-6 text-center text-uppercase mb-3"
    >
      <h1
        className="text-slanted"
      >
         recipe list 
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-1m mx-auto col-md-6 col-lg-4 my-3"
    >
      <div
        className="card"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <img
          alt="img"
          className="img-card-top"
          id="47746"
          src="http://forkify-api.herokuapp.com/images/best_pizza_dough_recipe1b20.jpg"
          style={
            Object {
              "height": "14rem",
            }
          }
        />
        <div
          className="card-body text-capitalize"
        >
          <h6>
            Best Pizza Dough Ever
          </h6>
          <h6
            className="text-warning text-slanted"
          >
            provided by 101 Cookbooks
          </h6>
        </div>
        <div
          className="card-footer"
        >
          <a
            className="btn btn-primary text-capitalize"
            href="#/recipes/47746"
          >
            details
          </a>
          <a
            className="btn btn-success mx-2 text-capitalize"
            href="http://www.101cookbooks.com/archives/001199.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            recipe url
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
